query.go

```diff
@@ -420,26 +420,30 @@ func (q *query) queryPeer(ctx context.Context, ch chan<- *queryUpdate, p peer.ID
 	ctx, span := internal.StartSpan(ctx, "IpfsDHT.QueryPeer")
 	defer span.End()
 
-	dialCtx, queryCtx := ctx, ctx
```

Comment:
Why can we not keep these contexts? I don't understand the benefit of using only `ctx`, especially if the behavior becomes racy.

Reply:
It is already racy, the variables do nothing they are never assigned or anything.

Reply:
The race is that if we fail to dial a peer but during the time we return the context is canceled we will incorrectly assume we failed to dial the peer because the context expired and not remove it from the routing table, refresh takes care of doing this properly so it's fine.

Reply:
Dropped